packages/entity/src/internal/CompositeFieldHolder.ts

```diff
@@ -0,0 +1,208 @@
+  }
+
+  getDataManagerDataLoaderKey(): string {
+    return this.serialize();
+  }
+
+  serializeLoadValue(
+    value: CompositeFieldValueHolder<TFields, EntityCompositeField<TFields>>,
+  ): SerializedCompositeFieldValueHolder {
+    return value.serialize();
+  }
+
+  deserializeLoadValue(
+    value: SerializedCompositeFieldValueHolder,
+  ): CompositeFieldValueHolder<TFields, EntityCompositeField<TFields>> {
+    return CompositeFieldValueHolder.deserialize(value);
+  }
+
+  validateRuntimeLoadValuesForDataManagerDataLoader(
+    values: readonly CompositeFieldValueHolder<TFields, EntityCompositeField<TFields>>[],
+    entityClassName: string,
+  ): void {
+    const isInvalidRuntimeCompositeFieldValue = (
+      value: CompositeFieldValueHolder<TFields, EntityCompositeField<TFields>>,
+    ): boolean =>
+      Object.values(value.compositeFieldValue).some(
+        (fieldValue) => fieldValue === undefined || fieldValue === null,
+      );
+
+    const invalidValueIndex = values.findIndex(isInvalidRuntimeCompositeFieldValue);
+    if (invalidValueIndex >= 0) {
+      throw new Error(
+        `Invalid load: ${entityClassName} (${String(this.compositeField)} = ${
+          values[invalidValueIndex]
+        })`,
+      );
+    }
+  }
+
+  vendNewLoadValueMap<V>(): LoadValueMap<
+    SerializedCompositeFieldValueHolder,
+    CompositeFieldValueHolder<TFields, EntityCompositeField<TFields>>,
+    V
+  > {
+    return new CompositeFieldValueHolderMap();
+  }
+}
+
+declare const CompositeFieldValueHolderSerializedBrand: unique symbol;
+export type SerializedCompositeFieldValueHolder = string & {
+  readonly [CompositeFieldValueHolderSerializedBrand]: true;
+};
+
+/**
+ * A load value for a CompositeFieldHolder.
+ */
+export class CompositeFieldValueHolder<
+  TFields extends Record<string, any>,
+  TCompositeField extends EntityCompositeField<TFields>,
+> implements IEntityLoadValue<SerializedCompositeFieldValueHolder>
+{
+  constructor(
+    public readonly compositeFieldValue: EntityCompositeFieldValue<TFields, TCompositeField>,
+  ) {}
+
+  toString(): string {
+    return `CompositeFieldValue(${Object.entries(this.compositeFieldValue)
+      .map(([fieldName, fieldValue]) => `${fieldName}=${fieldValue}`)
+      .join(',')})`;
+  }
+
+  public getFieldSet(): ReadonlySet<keyof TFields> {
+    return new Set(Object.keys(this.compositeFieldValue) as (keyof TFields)[]);
+  }
+
+  public static deserialize<
+    TFields extends Record<string, any>,
+    TCompositeField extends EntityCompositeField<TFields>,
+  >(
+    serialized: SerializedCompositeFieldValueHolder,
+  ): CompositeFieldValueHolder<TFields, TCompositeField> {
+    return new CompositeFieldValueHolder(
+      JSON.parse(serialized) as EntityCompositeFieldValue<TFields, TCompositeField>,
+    );
+  }
+
+  public serialize(): SerializedCompositeFieldValueHolder {
+    return JSON.stringify(
+      this.compositeFieldValue,
+      Object.keys(this.compositeFieldValue).sort(),
```

Comment:
This call doesn't look quite right.

Reply:
I think this is the first time I've seen the replacer used that way instead of as a function. It's a good use case, thanks for adding the comment.